src/common/AndroidUtils.ts

```diff
@@ -74,237 +72,216 @@ export class AndroidSDKUtils {
+            )
+            .catch((error) => Promise.reject(error));
     }
 
     public static async fetchAndroidSDKPlatformToolsLocation(): Promise<
         string
     > {
-        return new Promise(async (resolve, reject) => {
-            if (!AndroidSDKUtils.getAndroidSdkRoot()) {
-                return reject(new Error('Android SDK root is not set.'));
-            }
-            try {
-                CommonUtils.executeCommandSync(
-                    `${AndroidSDKUtils.getAdbShellCommand()} --version`
-                );
-                resolve(AndroidSDKUtils.getAndroidPlatformTools());
-            } catch (err) {
-                reject(err);
-            }
-        });
+        if (!AndroidSDKUtils.getAndroidSdkRoot()) {
+            return Promise.reject(new Error('Android SDK root is not set.'));
+        }
+
+        return CommonUtils.executeCommandAsync(
+            `${AndroidSDKUtils.getAdbShellCommand()} --version`
+        )
+            .then((result) =>
+                Promise.resolve(AndroidSDKUtils.getAndroidPlatformTools())
+            )
+            .catch((error) => Promise.reject(error));
     }
 
     public static async fetchInstalledPackages(): Promise<AndroidPackages> {
         if (!AndroidSDKUtils.getAndroidSdkRoot()) {
             return Promise.reject(new Error('Android SDK root is not set.'));
         }
 
-        if (!AndroidSDKUtils.isCached()) {
-            try {
-                const stdout = CommonUtils.executeCommandSync(
-                    `${AndroidSDKUtils.getSdkManagerCommand()} --list`
-                );
-                if (stdout) {
+        if (AndroidSDKUtils.isCached()) {
+            return Promise.resolve(AndroidSDKUtils.packageCache);
+        }
+
+        return CommonUtils.executeCommandAsync(
+            `${AndroidSDKUtils.getSdkManagerCommand()} --list`
+        )
+            .then((result) => {
+                if (result.stdout && result.stdout.length > 0) {
                     const packages = AndroidPackages.parseRawPackagesString(
-                        stdout
+                        result.stdout
                     );
                     AndroidSDKUtils.packageCache = packages;
                 }
-            } catch (err) {
-                return Promise.reject(err);
-            }
-        }
-
-        return Promise.resolve(AndroidSDKUtils.packageCache);
-    }
-
-    public static fetchEmulators(): Promise<AndroidVirtualDevice[]> {
-        return new Promise((resolve, reject) => {
-            let devices: AndroidVirtualDevice[] = [];
-            try {
-                const result = CommonUtils.executeCommandSync(
-                    AndroidSDKUtils.getAvdManagerCommand() + ' list avd'
-                );
-                if (result) {
+                return Promise.resolve(AndroidSDKUtils.packageCache);
+            })
+            .catch((error) => Promise.reject(error));
```

Comment:
Not trying to boil the ocean here, but just throwing it out as more forward thinking food for thought around best practices: should we ditch `catch` blocks like this, that simply propagate the caught error? I don't believe they're necessary (except in `new Promise()` constructs. 😂)

Reply:
➕

Reply:
Yes you're right. I'll clean these up.